cheatsheets_excluded/Content_Security_Policy_Cheat_Sheet.md

```diff
@@ -1,203 +1,199 @@
+  - `Content-Security-Policy: frame-ancestors trusted.com;`
 
-`The frame-ancestors directive MUST be ignored when monitoring a policy, and when a contained in a policy defined via a meta element.`
+## Loading Files from CDN
 
-In otherwords, this will not work when CSP is in a <meta> tag, and will not work when using Content-Security-Policy-Report-Only.
+The below CSP allows loading from the same origin and loading images and scripts from the CDN:
 
-When a report is generated, the blocked-uri will only have a value if it is the same origin as the page.
+`Content-Security-Policy: default-src 'self'; image-src cdn.example.com; script-src cdn.example.com;`
 
-Refactoring inline code
-=======================
+# Refactoring inline code
 
 By default CSP disables any unsigned JavaScript code placed inline in the HTML source, such as this:
 
+```js
 <script>
 var foo = "314"
 
 <script>
+```
+
 The inline code can be enabled by **specifying its SHA256 hash** in the CSP header:
 
-`    Content-Security-Policy: script-src 'sha256-gPMJwWBMWDx0Cm7ZygJKZIU2vZpiYvzUQjl5Rh37hKs='`
+> `Content-Security-Policy: script-src 'sha256-gPMJwWBMWDx0Cm7ZygJKZIU2vZpiYvzUQjl5Rh37hKs=';`
 
 This particular script's hash can be calculated using the following command:
 
-`   echo -n 'var foo = "314"' \| openssl sha256 -binary \| openssl base64`
+> `echo -n 'var foo = "314"' | openssl sha256 -binary | openssl base64`
 
 Some browsers (e.g. Chrome) will also display the hash of the script in JavaScript console warning when blocking an unsigned script.
 
-The inline code can be also simply moved to a separate JavaScript file:
-
-<script>
-var foo = "314"
-
-<script>
-becomes:
+The inline code can be also simply moved to a separate JavaScript file and the code in the page becomes:
 
+```js
 <script src="app.js">
 </script>
-with \`app.js\` containing the \`var foo = "314"\` code.
-
-The inline code restriction also applies to **inline event handlers**, so that the following construct will be blocked under CSP:
-
-`   `<button id="button1" onclick="doSomething()">
+```
 
-This should be replaced by \`addEventListener' calls:
+with `app.js` containing the `var foo = "314"` code.
 
-`   document.getElementById("button1").addEventListener('click', doSomething);`
-
-Variable assignment in inline scripts. Rather than do this:
-
-<script>
-var foo = "314";
-
-<script>
-Leverage HTML5's custom data attributes by setting the value as follows:
+The inline code restriction also applies to `inline event handlers`, so that the following construct will be blocked under CSP:
 
-`<body data-foo="314”>`
-`   ...`
+> `<button id="button1" onclick="doSomething()">`
 
-</body>
-And access the value by doing:
+This should be replaced by `addEventListener` calls:
 
-`  var itemID = document.body.getAttribute("data-foo”);`
+> `document.getElementById("button1").addEventListener('click', doSomething);`
 
-Authors and Primary Editors
-===========================
+# References
```

Comment:
Add ref to the MDN section about [CSP](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy) because it's more practical to use for a developer than the RFC...